glfs.c

```diff
@@ -412,7 +412,7 @@ static glfs_t* tcmu_create_glfs_object(char *config, gluster_server **hosts)
 	gluster_cache_refresh(fs, config);
 
  fail:
-	gluster_free_server(&entry);
+	gluster_free_server(hosts);
```

Comment:
Patch looks ok for the specific bug you are fixing. Will merge.

Just a warning. If we keep the check_config code, glfs_check_config does not init hosts, so when it calls tcmu_create_glfs_object if parse_imagepath fails before we do the

*hosts = calloc(1, sizeof(gluster_server));

then when we call gluster_free_server here it could be on a host pointer that points to whatever and not NULL like in the other code paths.